pandas/indexes/range.py

```diff
@@ -46,20 +47,17 @@ def __new__(cls, start=None, stop=None, step=None, name=None, dtype=None,
 
         # RangeIndex
         if isinstance(start, RangeIndex):
-            if not copy:
-                return start
```

Comment:
I don't follow you on ``.view``. I follow you on ``_shallow_copy``... which is equivalent to what I do, and does not depend on ``copy``, because there is nothing "deep" to copy.